### ISSUE TITLE
Weird results in dev and test set based on tf

### ISSUE THREAD

**Issue by nlpjoe** [allenai/bilm-tf#121]
I trained bilm model using all data(train+test) with token embedding and without char cnn. After convergence，I use it in a text classify model,textcnn, as a feature.

I use 5-fold cross validation as dev result, I got an accuracy of 75%, but in the testset I got an accuracy of 60%. I don't know why.

One of my colleagues has the same problem in QA field. How can I check the bug?

**Comment by matt-peters** [comment]
Seems like you are overfitting the test set.  I suggest following standard practice and only training on the train set, using the dev set to monitor heldout accuracy then stopping training when the dev accuracy begins to decrease.
